cypress/integration/productsTests.js

```diff
@@ -1,67 +1,91 @@
 import uuidv4 from "uuid/v4";
 
-describe('CRUD products tests', function() {
-    let testUserMail;
-    let testPwd;
+import { getTestConfig } from "../config";
 
-    before(function() {
-        cy.getTestData().then(($result) => {
-            testUserMail = $result.testUserMail;
-            testPwd = $result.testPwd;
-        });
-    })
+describe("CRUD products tests", () => {
+  const { testUserMail, testPwd } = getTestConfig();
 
-    beforeEach(function() {        
-        cy.reLogin(testUserMail, testPwd);
-    });
+  beforeEach(() => {
+    cy.reLogin(testUserMail, testPwd);
+  });
 
-    //To keep the balance, I create 2 products because 2 products are deleted in following tests
-    it('Create 2 new products', () => {
-        cy.navigateToProductsPage();
-        cy.createTestProduct();
-        cy.createTestProduct();
-    });
+  // To keep the balance, I create 2 products because 2 products are deleted in following tests
+  it("Create 2 new products", () => {
+    cy.navigateToProductsPage();
+    cy.createTestProduct();
+    cy.createTestProduct();
+  });
 
+  // We're expecting this test to fail once pagination of tables gets implemented
+  it("'Delete product' removes one row from table", () => {
+    cy.navigateToProductsPage();
+    cy.get("tbody[data-cy=productsTableBody")
+      .find("tr")
+      .its("length")
+      .then(rowsBefore => {
+        cy.get("button[data-cy=deleteProductButton]")
+          .first()
+          .click();
+        cy.get("div[data-cy=deleteConfirmationDialog").should("be.visible");
+        cy.get("button[data-cy=confirmDeleteButton").click();
+        // cy.wait(2000); // updating table takes some time
```

Comment:
@DurkoMatko the cypress linting rules flag this as a 'bad practice' - not sure if there is an alternative way to get it working. I'll take a look once I've figured out the issues on the existing PR with these tests

Reply:
Yeah it should be done using `.then(()=>{})` which resolves first when the click event finishes all it's "effects"...but it didn't work, the table was still in the old state. I can look at it again but I've actually spend quite some time on it and decided to go just with wait() as it seemed to work fine and I didn't want to loose more time.